### PR TITLE
Replace is_ip_address with Puppet 4 native comparision

### PR DESCRIPTION
**Pull request by baurmatt**

### CODE REVIEW

**Comment by baurmatt** [comment]
Same error as in #553. I'm pretty confident that this has nothing to do with this change.

File: manifests/proxy.pp [hunk 1]
```diff
@@ -429,7 +429,7 @@
   if ($listenip != undef) {
     if ($listenip =~ /^(eth|lo|bond|lxc|eno|tap|tun).*/) {
       $listen_ip = getvar("::ipaddress_${listenip}")
-    } elsif is_ip_address($listenip) {
+    } elsif $listenip =~ Stdlib::Compat::Ip_address {
```
Comment (alexjfisher) [inline manifests/proxy.pp]: Should this use `Stdlib::IP::Address` instead of `Stdlib::Compat::Ip_address`?

Comment (Dan33l) [reply manifests/proxy.pp]: `metadata.json` requires `stdlib >= 4.19.0 < 6.0.0`.
Data type `Stdlib::IP::Address` requires  at least `stdlib 4.25.0`.

Comment (baurmatt) [reply manifests/proxy.pp]: @Dan33l That's the reason why I chose `Stlib::Compat::Ip_address` :) If its ok to increase the stdlib dependency to 4.25.0, I'm happy to use `Stdlib::IP::Address`.

Comment (alexjfisher) [reply manifests/proxy.pp]: I'd be fine with increasing the stdlib dependency.

**Comment by alexjfisher** [comment]
@baurmatt Test failure is fixed in https://github.com/voxpupuli/puppet-zabbix/pull/557 (which should be merged shortly)

**Comment by baurmatt** [comment]
@alexjfisher Thanks, rebased from master! :)

**Comment by alexjfisher** [comment]
@baurmatt Thanks.  I had actually already restarted your build, (and when travis builds a PR, it builds the candidate merge commit)

**Pull request merged**